File: Java-Core/src/main/java/presto/value/DateRange.java
package presto.value;

import org.joda.time.LocalDate;

import presto.type.DateType;



public class DateRange extends Range<Date> {

	public DateRange(Date left, Date right) {
		super(DateType.instance(), left, right);
	}
	
	@Override
	public long length() {
		long h = high.getValue().toDate().getTime();
		long l = low.getValue().toDate().getTime();
		return 1 + ( (h-l)/(24*60*60*1000));
	}

	@Override
	public int compare(Date o1, Date o2) {
		return o1.compareTo(o2);
	}

	@Override
	public Date getItem(long index) {
		LocalDate result = low.getValue().plusDays((int)index-1);
		if(result.isAfter(high.getValue()))
			throw new IndexOutOfBoundsException();
		return new Date(result);
	}
	
	@Override
	public Range<Date> newInstance(Date left, Date right) {
		return new DateRange(left, right);
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/value/DateRange.js
var Range = require("./Range").Range;
var LocalDate = require("./LocalDate").LocalDate;
var DateType = null;

exports.resolve = function() {
    DateType = require("../type/DateType").DateType;
};

function DateRange(left, right) {
	Range.call(this, DateType.instance, left, right);
	return this;
}

DateRange.prototype = Object.create(Range.prototype);
DateRange.prototype.constructor = DateRange;


DateRange.prototype.size = function() {
	var h = this.high.value.valueOf();
	var l = this.low.value.valueOf();
	return 1 + ( (h-l)/(24*60*60*1000));
};

DateRange.prototype.getItem = function(index) {
	var millis = this.low.value.valueOf() + (index-1)*(24*60*60*1000);
	if(millis>this.high.value.valueOf()) {
		throw new IndexOutOfBoundsException();
	} else {
		return new LocalDate(new Date(millis));
	}
};

/*
@Override
public Range<Date> newInstance(Date left, Date right) {
	return new DateRange(left, right);
}

*/


exports.DateRange = DateRange;
